ntbs-service/Models/Validations/ValidationAttributes.cs

```diff
@@ -44,4 +45,47 @@ private bool isTruthy(object value)
             return value != null && (bool)value;
         }
     }
+
+    public class ValidPartialDateCanConvertToDatetimeAttribute : ValidationAttribute
+    {
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            PartialDate valueCopy = (PartialDate) value;
+            if(valueCopy == null || valueCopy.IsEmpty()) {
+                return null;
+            }
+            if (!string.IsNullOrEmpty(valueCopy.Day)) {
+                if(string.IsNullOrEmpty(valueCopy.Month) || string.IsNullOrEmpty(valueCopy.Year)) {
+                    return new ValidationResult("Year and month must be provided if a day has been provided");
+                }
+            }
+            if(!string.IsNullOrEmpty(valueCopy.Month)) {
+                if(string.IsNullOrEmpty(valueCopy.Year)) {
+                    return new ValidationResult("A year must be provided");
+                }
+            }
+            bool canConvert = valueCopy.TryConvertToDateTimeRange(out DateTime? x, out DateTime? y);
+            if (!canConvert) {
+                return new ValidationResult(ErrorMessage);

```

Comment:
do we need a default if the client doesn't provide an `ErrorMessage`? or is it required?